tests/suites/test_suite_psa_crypto_driver_wrappers.function

```diff
@@ -0,0 +1,243 @@
+/* BEGIN_HEADER */
```

Comment:
Thinking about it, it seems to me that we don't really need that. This is testing test code and the test code is PSA code or mbedtls code anyway (thus already tested) . I think it would be more useful to focus on amending/improving the PSA tests to check that when an operation (based on C pre-processor defines) is supposed to be done by a test driver we check that the test driver is called as expected (check of the call and check of the call parameters). If there is no equivalent test to the signature tests done here, it would be great to add them to the PSA tests.

Reply:
I don't understand your comment: we don't really need what?

There are several uses for test drivers:

* Plug in transparent drivers that are expected to work normally and run the normal tests.
* Plug in opaque drivers that are expected to work normally and possibly have extra instrumentation (e.g. to verify that the driver was called when it should have), and run dedicated tests that support opaque drivers (which most don't).
* Plug in transparent drivers that are expected to work normally and have extra instrumentation and run dedicated tests that exercise the extra instrumentation.
* Plug in specially crafted drivers that can be instructed to misbehave in certain ways (e.g. returning an error), and check that the core handles the misbehavior correctly, from dedicated tests.

`test_suite_psa_crypto_driver_wrappers.function` gets started on the dedicated tests that will validate the wrapper code, validating properties such as making sure that the wrapper code calls the correct entry point.

Reply:
Sorry, I mixed up what was done in #3605 with what is done here. The tests below call the PSA API (not the test driver entry points as I was thinking when writing my comment above) and check the dispatch to the test driver which is indeed something we want to test. As I mentioned above this could be done by amending the PSA tests but this would probably make them too complex. To make it short, just forget about my comment above and sorry for the noise.

Reply:
Thanks. Can you close/resolve this then?

Reply:
Indeed.